### ISSUE TITLE
Pre-work Checklist: DevOps-Security-Member: drakeredwind01

### ISSUE THREAD

**Issue by drakeredwind01** [hackforla/devops-security#36]
### Prerequisite
We are looking forward to having you on our team. Please ensure you have prior experience with the HfLA website team before contributing to our repository.
### Overview
As a new member on the HfLA devops-security team, fill in the following fields as you complete each onboarding item.

### Special Notes
1. Keep this issue open until you've completed all steps, including learning to provide updates for longer tasks.
2. Normally, handle one issue at a time, but this one is an exception as it teaches how to manage multiple tasks.
3. Work on action items sequentially, but proceed if possible. For example, set up your dev environment without waiting for the weekly meeting.

### Action Items

- [x] Before starting to work on the below instructions, make sure to join the #ops Slack Channel. And are a member of `devops-security` repository.
- [x] Self-assign this issue (gear in right side panel).
- [x] Add this issue to the Project Board under the Projects section (gear in right side panel).
- [x] Attend weekly team meeting, Wednesdays 6-8pm PST.
  - [x] Note: There are no meetings on the 1st-7th of every month.
- [x] Complete the steps in [Creating a personal AWS account](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#creating-a-personal-aws-account) and [Login as root user & setup MFA](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#login-as-root-user-&-setup-mfa).
- [x] Read and follow the instructions in [Setting up IAM and AWS CLI](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#setting-up-iam-and-aws-cli) for:
    - [x] [Creating an IAM User](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#create-an-iam-group)
    - [x] [Creating an IAM Group](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#create-an-iam-group)
    - [x] [Attaching IAM user to IAM Group](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#attach-iam-user-to-iam-group)
    - [x] [Providing `AdministratorAccess` policy to IAM group](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#attach-administratoraccess-policy-to-iam-group)
    - [x] Log in as the newly created user instead of continuing to log in as the root user (it is not recommended to login with root access).
    - [x] [Generating user access keys](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#generating-access-keys-for-aws-cli)
- [x] Complete the instructions in [AWS Documentation](https://docs.aws.amazon.com/cli/v1/userguide/cli-chap-install.html) and choose your operating system to install AWS CLI. 
- [x] Complete the instruction in [AWS Documentation](https://docs.aws.amazon.com/cli/latest/userguide/cli-authentication-short-term.html) to setup the AWS CLI.
- [x] Read follow the instructions in [Creating a backend state](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#creating-backend-state).
- [x] Install Terraform locally by following the instructions of the installation guide mentioned in [Installing Terraform](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#installing-terraform)
- [ ] Install Terraform Docs locally by following the instructions of the installation guide mentioned in [Installing Terraform docs](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#installing-terraform-docs)
- [x] Complete the instructions in [Clone the repository](https://github.com/hackforla/devops-security/blob/main/CONTRIBUTING.md#clone-the-repository)
- [x] Submit a [new request](https://github.com/hackforla/devops-security/issues/new?assignees=&labels=enhancement&projects=&template=request-aws-iam-resources.yml) to create new AWS user account and then self-assign this issue.
- [x] Create a new branch from main by executing the command

    ```bash
    git checkout -b issue-number-add-new-iam-user

    ```
- [x] Navigate to the `aws-user.tf` file and add your user information and follow the below template.

    ```bash
    
    module "iam_user_testiamuser" {
    source = "./modules/aws-users"

    user_name = "testiamuser"
    user_tags = {
      "Project"      = "devops-security"
      "Access Level" = "1"
    }
    user_groups = ["read-only-group"]
    }

    ```
- [ ] In your code editor navigate to `terraform` directory. `cd terraform`
- [ ] Execute the command `terraform init` to initialize terraform in the directory. Address any failures that arise (if any).
- [ ] Execute the command `terraform plan` this will output a plan replicating the same IAM resources as the devops security account. Address any failures that arise (if any).
- [ ] Then execute the command `terraform apply` this will create all of the resources that are currently managed by Devops Security. All of the resources created here incur zero cost except for the Dynamo DB installation, which should remain in the free tier.
      - [ ] ** If you have cost concerns, Run a Terraform Destroy to take down all of the resources you created (don't worry, you can recreate them just as quickly). **
- [ ] Once you have tested your changes, stage them in git with 
    - [ ] `git status` command.
    - [ ] then `git add path/to/file` (you can copy from above output for the file path).
- [ ] Commit the changes by executing `git commit -m "briefly describing the changes"`.
- [ ] Push the changes with `git push --set-upstream origin name-of-branch`

**Comment by drakeredwind01** [comment]
prerequisites needs a new title and the template needs real prerequisites 
### intro
We are looking forward to having you on our team. Please ensure you have prior experience with the HfLA website team before contributing to our repository.
### prerequisites 
- [ ] need 1
- [ ] need 2
- [ ] need 3

**Comment by ExperimentsInHonesty** [comment]
@drakeredwind01 please make a new issue that outlines the problem like you have done in this issue... but that is specific to the changes that you are suggesting.  that way, we can give your issue (this one) back to you and still have an issue specifically for updating the prework template.

**Comment by drakeredwind01** [comment]
Thank you for your advice. I believe it may be easier (after looking at it) to finish all the changes first and then create the separate issues from the derivatives, as I’m constantly restructuring the list.

Please let me know if you would prefer to have play-by-play issues instead of final, concrete suggested changes and an overall example, which is what this current method would ultimately provide. I can see the merits of both methods.

**Comment by ExperimentsInHonesty** [comment]
@drakeredwind01 please document all the changes you are suggesting in a comment on this issue.

**Comment by drakeredwind01** [comment]
My problem
I get error 
╷
│ Error: Required plugins are not installed
│
│ The installed provider plugins are not consistent with the packages selected in the dependency lock file:
│   - registry.terraform.io/hashicorp/aws: there is no package for registry.terraform.io/hashicorp/aws 5.34.0 cached in .terraform\providers
│
│ Terraform uses external plugins to integrate with a variety of different infrastructure services. To download the
│ plugins required for this configuration, run:
│   terraform init
╵
I think the important part is:
│   - registry.terraform.io/hashicorp/aws: there is no package for registry.terraform.io/hashicorp/aws 5.34.0 cached in .terraform\providers

Possible solution:
https://github.com/gruntwork-io/terragrunt/issues/1960
It seems they fixed this by deleting “.terraform.lock.hcl”

Can confirm removing the lock files solves it, but you need to do it on the current module and any it depends on.
terragrunt version v0.36.1
Terraform v1.1.5
on darwin_amd64

Itsayellow says:
He also had this issue and deleting the terraform lock files but that wasn't helping.
But “Finally I found if I deleted ./.terragrunt-cache from where I was executing terragrunt, it finally worked without error for me.”

Extra advice:
luogedai commented on Jan 13, 2022
If you have dependencies, try to clear and init or upgrade the dependencies first. This step solved my issue.

AWS Support plans
https://us-east-1.console.aws.amazon.com/support/plans/home#/
https://aws.amazon.com/premiumsupport/pricing/

**Comment by sudhara** [comment]
@drakeredwind01 

Please provide update
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by drakeredwind01** [comment]
Just complexly finished the prework checklist and altered the "aws-users.tf" file and pushed, and created a merge request for approval successfully.

I am also searching for more CloudShell commands to integrate a more programmatic process for newer members to enhance the cohesion process.

**Comment by drakeredwind01** [comment]
2. blockers
the blockers have been systematicly anihilated and accounted for in the new tut I created for a smooth induction process.

3. my time is highly variable in this uncertain time in my life. I have already devote much time into this singular process and will continue to work on the process to enhance the new user experience.

4. finished

5. link
link for the pull request:     https://github.com/hackforla/devops-security/pull/83
